lib/resque/plugins/status.rb

```diff
@@ -243,8 +268,49 @@ def kill!
         raise Killed
       end
 
+      # Initiates parent once total number of child jobs is known
+      # This step is essential to prevent race condition of all currently queued children finish,
+      # while parent still plans to enqueue more.
+      # When child job completes, it increments the `num` of the parent job by 1.
+      # When `num` gets == `total` the parent job marked as complete and `on_success` is called on the last child job
+      # If parent is killed, all children are prevented from running
+      # Child job statuses are deleted when complete or killed to avoid garbage in redis. It's preserved on error.
+      def init_parent(total)
+        at(0, total, "Queuing #{total} subjobs")
```

Comment:
If you run jobs that have the same execution time then they will finish simultaneously. So you will see no progress during execution

Reply:
The only this could happen is if the jobs are only a few and very long so they all able to run in parallel and finish at the same time. This is not a very good usage of the system.
The system designed for many short child jobs. Potentially thousands. It's not a data pipeline orchestrator, it's distributed job runner. So when you split the large job into child jobs, split it into jobs that ideally run for up to 1m. The number of jobs almost doesn't matter (10k is fine, 100k not so much), as long as it's running fast.